src/Sextant.WinUI/NavigationView.xaml.cs

```diff
@@ -0,0 +1,398 @@
+                        if (resetStack)
+                        {
+                            _mirroredPageStack.Clear();
+                            _mirroredPageStack.Push(viewModel);
+
+                            mainFrame.Navigate(pageType, null, new SuppressNavigationTransitionInfo());
+                            if (mainFrame.Content is IViewFor viewForReset)
+                            {
+                                viewForReset.ViewModel = viewModel;
+                            }
+                            else
+                            {
+                                var contentTypeName = mainFrame.Content?.GetType().ToString() ?? "Unknown Type";
+                                _logger.Debug(
+                                    $"The view ({contentTypeName}) does not implement IViewFor<>.  Cannot set ViewModel of type, {viewModel.GetType()}, on view.");
+                            }
+
+                            mainFrame.BackStack.Clear();
+
+                            return Observable.Return(Unit.Default);
+                        }
+
+                        NavigationTransitionInfo animation;
+                        if (animate)
+                        {
+                            animation = new EntranceNavigationTransitionInfo();
+                        }
+                        else
+                        {
+                            animation = new SuppressNavigationTransitionInfo();
+                        }
+
+                        _mirroredPageStack.Push(viewModel);
+
+                        mainFrame.Navigate(pageType, null, animation);
+                        if (mainFrame.Content is IViewFor viewFor)
+                        {
+                            viewFor.ViewModel = viewModel;
+                        }
+                        else
+                        {
+                            var contentTypeName = mainFrame.Content?.GetType().ToString() ?? "Unknown Type";
+                            _logger.Debug(
+                                $"The view ({contentTypeName}) does not implement IViewFor<>.  Cannot set ViewModel of type, {viewModel.GetType()}, on view.");
+                        }
+
+                        return Observable.Return(Unit.Default);
+                    });
+
+        private static Type LocatePageTypeFor(object viewModel, string? contract)
+        {
+            var uwpViewTypeResolver = Locator.Current.GetService<ViewTypeResolver>(contract);
+
+            var viewType = uwpViewTypeResolver?.ResolveViewType(viewModel.GetType());
+
+            if (viewType is null)
+            {
+                throw new InvalidOperationException(
+                    $"No view could be located for type '{viewModel.GetType().FullName}', contract '{contract}'. Be sure Splat has an appropriate registration.");
+            }
+
+            return viewType;
+        }
+
+        private Page LocatePageFor(object viewModel, string? contract)
```

Comment:
Why is there `LocatePageTypeFor` and `LocatePageFor`?
One is used by `PushModal`, the other one by `PushPage`.
Based on the name are supposed to do the same thing, except they aren't.
Different view resolver and `LocatePageFor` has side effects, too.

Can this unified?
Why is there are `RegisterWinUIViewLocator` beside the default one?
The original UWP comments indicate to "return UWP views only" but why?